src/Makefile

```diff
@@ -20,7 +20,7 @@ DEPENDENCY_TARGETS=hiredis linenoise lua
 NODEPS:=clean distclean
 
 # Default settings
-STD=-std=c11 -pedantic -DREDIS_STATIC=''
+STD=-std=c99 -pedantic -DREDIS_STATIC=''
```

Comment:
I thought the problem was with the incorrect implementation of atomics, not the c11 part. So shouldn't we leave this as is and use the new redisAtomic's for systems that don't have _atomic?

Reply:
@madolson What was "incorrect" with the _Atomic? 
I think we wanna get back to C99 and be very portable. 
It's true that some systems have compilers that support C11 and not _Atomic, but others don't support C11 at all. 
Also, leaving in the makefile would mean its hard to make sure some C11 feature doesn't creep in. 

@ShooterIT please remove the `scl` parts from the Centos tests in `.github/workflows/daily.yml` and see that it passes without them.

Reply:
I don't know what mistakes I make of implementation of atomics, please help me point errors. @madolson 

@oranagra copy that, let me test it.

Reply:
@oranagra My comment was mostly aimed at I don't want to maintain the C99 and C11 conventions if we don't have to. If we don't want C11 to creep in, I don't we should keep any references to C11 code around. (I still see _Atomic usage). I'm not clear why would keep it since the original benefit was reducing the amount of code we were maintaining.

Reply:
Some systems didn't implement the _atomic part of C11, which was the few issues I saw. There might be other issues we don't support it. I still think we should be building on C11 with _Atomics by default, and we are aware of the systems that it's not supported on and we fall back to C99. 

edit: It looks like we agree on that based on your comment below.

Reply:
Maybe one benefit is that there might be some platform that supports _Atomic and doesn't support either atomic-builtin or sync-builtin?

If that's not technically possible, then i agree we can trim the block that's using _Atomic and avoid mentioning `STD=-std=c11` in the read me or elsewhere. 
But IIUC these builtins are non standard so we can't be certain about it, right?

Reply:
@madolson I think this PR is to keep redis compatibility on different compiling environment. I am also not sure there is a platform that supports _Atomic and doesn't support either atomic-builtin or sync-builtin @oranagra But redis before 6.0 can already run on many platforms, maybe that means it is a relatively rare case. But I'm not certain about that.